### PR TITLE
port registry usage to native Node module

### LINKED ISSUE #3105: unable to detect PowerShell due to group policy blocking registry access
### Description

In the previous version of GitHub desktop client I could specify PowerShell as the shell to launch for a given project, and it gave me cool information in the command line about the current state of the repo (if I was ahead/behind, conflicts, changes, etc.). There is no longer an option for me to select PowerShell in the new version. There is a drop down in advanced options for shell, but it only has "Command prompt" in it.

### Version

**GitHub Desktop version:** 1.0.4

**OS version:** Microsoft Windows [Version 10.0.15063]

### Steps to Reproduce

1. Launch the client
2. Go to File -> Options and click on the Advanced tab
3. Click the drop down for Shell.

Expected: A number of shells are available (cmd, ps, bash, etc.)
Actual: only cmd is available.

4. Launch the cmd shell for a project.
Expected: it gives data about the local branch vs. tracking branch in the command line
Actual: it does not.

**Expected behavior:** That I can use PowerShell for git through github desktop client and that it gives me information in the shell about the branch.

**Actual behavior:** It is not possible and I can only use a command prompt with no info.

**Reproduces how often:** 100%

### LINKED ISSUE #3405: App shows only a white screen when trying to see Options
### Description

When I go to File > Options a modal is briefly displayed before the whole screen turns white.

### Version

1.0.9 (Windows 10.0.15063)

**GitHub Desktop version:** 1.0.9

Windows 10.0.15063

### Steps to Reproduce

1. Open GitHub desktop
2. Click File
3. Click Options
4. White screen

**Expected behavior:** To see the Options dialog

**Actual behavior:** The options dialog appears for a fraction of a second before the whole app screen goes white

**Reproduces how often:** Every time, I've never been able to see the options screen.

### Logs

The logs only show an app launch message.

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #3105
Fixes #3405

Supersedes #3209 

- [x] port registry usage over to new library
- [x] update `package.json` so that non-`win32` platforms can compile
- [x] test and benchmark performance differences (should be significantly quicker)
- [x] confirm `registry-js`returns UTF-8 strings (avoiding the need for any active code page shenanigans)

- [x] confirm app still works fine with group policy enabled to block registry editing tools
- [x] test on Windows 7 VM
  - [x] VSCode Insiders install (32-bit) found
  - [x] fresh install of Atom found
  - [x] PowerShell default path found
- [x] test on Windows 8 VM
  - [x] VSCode install (64-bit) found
  - [x] Hyper shell found
  - [x] PowerShell default path found
- [x] test on Windows 10 VM
  - [x] Sublime Text found
  - [x] Git Bash found
  - [x] PowerShell default path not found
- [x] cleanup changelog and version changes

### CODE REVIEW

File: app/src/lib/editors/win32.ts [hunk 1]
```diff
@@ -144,6 +185,14 @@ function isExpectedInstallation(
   }
 }
 
+function getKeyOrEmpty(
+  keys: ReadonlyArray<RegistryValue>,
+  key: string
+): string {
+  const entry = keys.find(k => k.name === key)
+  return entry && entry.type === RegistryValueType.REG_SZ ? entry.data : ''
```
Comment (shiftkey) [inline app/src/lib/editors/win32.ts]: For reference, `registry-js` does some lifting to map the registry value type to a JS type. I only implemented strings (`REG_SZ`) and number (`REG_DWORD`) for now, but we can add more in the future:

```
export type RegistryStringEntry = {
  readonly name: string
  readonly type: RegistryValueType.REG_SZ | RegistryValueType.REG_EXPAND_SZ
  readonly data: string
}

export type RegistryNumberEntry = {
  readonly name: string
  readonly type: RegistryValueType.REG_DWORD
  readonly data: number
}

// TODO: define some other shapes of data
export type RegistryValue = RegistryStringEntry | RegistryNumberEntry
```

**Comment by shiftkey** [comment]
After a few runs of `master` and this PR, the average time to execute a registry lookup through `reg.exe` is ~223ms. With this change, the execution goes down to essentially zero (the occasional operation takes ~4ms but that's minor...

**Comment by shiftkey** [comment]
@begoldsm @ian-hamlin as this is a rather big change with some relatively new code, I've made a test build for Desktop (Windows only) available [here](https://s3.amazonaws.com/github-desktop/releases/1.0.10-test3-0152ecb9/GitHubDesktopSetup.exe).

Are you able to test this out and confirm:

  - a) you can view the Options dialog, and
  - b) you can see a list of editors and shells that you are installed on your machine

**Comment by ian-hamlin** [comment]
@shiftkey This is working for me now, thanks!

> a) you can view the Options dialog

This dialog now opens with no white-screens.

> b) you can see a list of editors and shells that you are installed on your machine

I can see items as expected.

**Comment by shiftkey** [comment]
@ian-hamlin that's great to hear! I'll do some more testing across the other versions of Windows we support, hopefully we can wrap this up :soon:

**Comment by shiftkey** [comment]
Alright, this is ready for review.

I'm open to improvements around the `registry-js` API too, while we're in here.

**Pull request merged**